### ISSUE TITLE
AI(18th day)

### ISSUE THREAD

**Issue by sdfs1231** [sdfs1231/AI#18]
tensorflow使用feed_dict语法来给占位符填充内容

**Comment by sdfs1231** [comment]
Y = tf.add(tf.matmul(W, X), b)

**Comment by sdfs1231** [comment]
因为向量中只有一个元素是1，其它的都是0
改用tensorflow框架的话，只需要一行代码:
    tf.one_hot(indices, depth, axis)

**Comment by sdfs1231** [comment]
X = tf.placeholder(tf.float32, [n_x, None], name="X")
其中[n_x,None]这里的none就是不确定样本数量

**Comment by sdfs1231** [comment]
X = tf.placeholder(dtype=tf.float32, shape=[144, 10], name='X')
这里可以指定shape

**Comment by sdfs1231** [comment]
transpose表示装置操作，所以logits相当于Z3.T

**Comment by sdfs1231** [comment]
_ , c = sess.run([optimizer, cost], feed_dict={X: minibatch_X, Y: minibatch_Y})

但我们在编码是，如果不需要某个返回值，我们就用`_`来代替。上面代码中我们就不需要返回的第一个结果，只需要第二个结果c，它代表了cost成本值。

**Comment by sdfs1231** [comment]
# 因为这也是tensorflow的一个特性，它会在计算成本函数时顺带把a3给我们计算了，
    # 所以前向传播中我们只需要计算Z3就可以了。
计算comput_cost的时候

**Comment by sdfs1231** [comment]
在使用argmax()函数时，比如在深度学习里面计算acc经常要用到这个参数，这个参数返回的是沿轴axis最大值的索引值
